### DISCUSSION TITLE
Error with the global AD indexing configuration to run INSFV

### DISCUSSION THREAD

**Discussion by AVxAle** [idaholab/moose#17721] [Q&A Modules: Navier-Stokes] [answered]
Hello Moose community,

I am having an issue with the enabling the global AD indexing for the finite volume simulations and I am hoping to get your help on this.
Every time I try to run my input file I get the error I reported below. I ran the "./configure --with-ad-indexing-type=global"  as suggested but this did not solve the issue. The output after running this script is also shown below. I also tried recompiling moose after running this script, but this did not make a difference either.

The input file I am trying to run is one of the example files available in: https://mooseframework.inl.gov/modules/navier_stokes/fv.html

FYI, my Application Version is: snapshot-20-10-27-17866-ge8ca19424f

Thank you in advance

----- Printscreen from the input file -----
*** ERROR ***
INSFV is not supported by local AD indexing. In order to use INSFV, please run the configure script in the root MOOSE directory with the configure option '--with-ad-indexing-type=global'

----- Printscreen of "./configure --with-ad-indexing-type=global" -----
$ ./configure --with-ad-indexing-type=global
---------------------------------------------
----------- Configuring MOOSE ---------------
---------------------------------------------
configuring with derivative backing array size of 50
configuring with AD indexing type global
configuring with derivative type sparse
checking for pkg-config... /usr/bin/pkg-config
configure: checking system for libpng
configuring with libpng support
configure: creating ./config.status
config.status: creating conf_vars.mk
config.status: creating framework/include/base/MooseConfig.h.tmp
config.status: framework/include/base/MooseConfig.h.tmp is unchanged
config.status: executing framework/include/base/MooseConfig.h commands
config.status: creating framework/include/base/MooseConfig.h - prefix MOOSE for framework/include/base/MooseConfig.h.tmp defines
config.status: framework/include/base/MooseConfig.h is unchanged
---------------------------------------------
--------- Done Configuring MOOSE ------------
---------------------------------------------

**Comment by GiudGiud** [comment, marked as answer]
Hello

You do have to re-build moose after re-configuring it to use global AD indexing.
Did you re-build the navier stokes module / your application including the navier stokes module after running the configure script?
Can you please try again after running `make clobberall` then re-building it ?
Which object is returning the error? If it doesn't say you may build moose in debug mode with `METHOD=dbg make -j xx`

Best
Guillaume
